uwblueprint/focus-on-nature
# Edlp registration

## Notion ticket link

[Registrant Experience: Early drop-off late pick-up step](https://www.notion.so/uwblueprintexecs/Registrant-Experience-Early-drop-off-late-pick-up-step-4d80e94bacc847b8ac2d95b9636bda75)



## Implementation description
* Implemented EDLP registration that conditionally renders if checked on the Additional Info step of registration
* Created two layouts, one that shows on a certain screen width, with the forms referencing state values so that their selections persist if the user messes with their screen width
* Create function to compute the number of available EDLP times, the fees and store the selected times and the total fees
* HARDCODED: TWO CAMPERS AND THE EDLP COST TO 5 DOLLARS/30 MIN. 



## Steps to test
1. Use a camp that has the needed details ( I used http://localhost:3000/register/camp/63538bc01bf0453b8a55ea1a)
2. Test that the EDLP toggle shows and hides the accordions
3. Test that the accordions work, changing the times, computes the EDLP fees and and makes sense, such as selecting longer times gives more fees
4. Check that number totals shown are correct (it is hardcorded to have 2 campers)
5. Check that the EDLP time selections are persisted when the screen width changes between mobile and desktop layouts and that the layout is responsive



## What should reviewers focus on?
* The `AdditionalInfo` folder, particularly edlpSessionRegistration.tsx and the index.tsx


## Checklist
